src/StackExchange.Redis.Extensions.Core/StackExchangeRedisCacheClient.cs

```diff
@@ -628,8 +628,13 @@ public IEnumerable<string> SearchKeys(string pattern)
 
 			foreach (var endpoint in endPoints)
 			{
-				var dbKeys = Database.Multiplexer.GetServer(endpoint).Keys(Database.Database, pattern);
+				var server = Database.Multiplexer.GetServer(endpoint);
+				if (!server.IsConnected || !server.Features.Scan)

```

Comment:
We check for **server.IsConnected**: this check is also made internally but since the idea here is to iterate through various servers, having one offline shouldn't disrupt the entire operation.
We also check for **server.Features.Scan**: this check is important in the scenario where scanning of keys has been disallowed in one of the servers, in which case the entire operation shouldn't be disrupted either.